setup.py

```diff
@@ -2003,11 +2003,11 @@ def detect_ctypes(self, inc_dirs, lib_dirs):
             ffi_inc = find_file('ffi.h', [], inc_dirs)
         if ffi_inc is not None:
             ffi_h = ffi_inc[0] + '/ffi.h'
+            # Just check that the file is not empty
```

Comment:
Checking that the header exists and not empty is strange. If you install an empty ffi header failing the build seems like the best way.

I would check why the code checking for LIBFFI_H was added.

If we really want to know if there is a non empty file we can just do:

    os.path.getsize(ffi_h) > 0